src/subscription_manager/repofile.py

```diff
@@ -461,19 +461,28 @@ def sections(self):
 
         def fix_content(self, content):
             # Luckily apt ignores all Fields it does not recognize
-            baseurl = content["baseurl"]
+            parsed_url = urlparse(unquote(content["baseurl"]))
+            baseurl = parsed_url._replace(query="").geturl()
             url_res = re.match(r"^https?://(?P<location>.*)$", baseurl)
```

Comment:
TBH: if now `baseurl` is parsed as URL, then it'd be a good idea to replace this regexp to extract hostname/path/etc with the same `urllib.parse` APIs -- for example replacing the scheme to get the katello URL

no need to do it in the existing comment; either as additional commit on top of this, or as future PR/improvement

Reply:
I will have a look into trying this out (likely early next week).

Reply:
This is what I came up with for using `urllib` instead of a regex for parsing the URL: https://github.com/candlepin/subscription-manager/pull/3454/commits/980e694189b9912c64129556ddc704878fdbc74a

I do think this is somewhat more readable overall.

Reply:
After some trial and error I did add this change to the existing commit. Please let me know if I should separate it out again.